arangod/RocksDBEngine/RocksDBKeyBounds.cpp

```diff
@@ -458,25 +455,47 @@ RocksDBKeyBounds::RocksDBKeyBounds(RocksDBEntryType type, uint64_t first,
 
 /// iterate over the specified bounds of the velocypack index
 RocksDBKeyBounds::RocksDBKeyBounds(RocksDBEntryType type, uint64_t first,
-                                   VPackSlice const& second, VPackSlice const& third)
+                                   VPackSlice const& second, VPackSlice const& third,
+                                   bool isReverse)
     : _type(type) {
   switch (_type) {
     case RocksDBEntryType::VPackIndexValue:
     case RocksDBEntryType::UniqueVPackIndexValue: {
-      size_t startLength = sizeof(uint64_t) + static_cast<size_t>(second.byteSize());
-      size_t endLength = 2 * sizeof(uint64_t) + static_cast<size_t>(third.byteSize());
+      uint64_t objectIdStart = first;
+      size_t byteSizeStart = 0;
+      if (isReverse && second.isArray() && second.length() == 1 && second.at(0).isMinKey() && objectIdStart > 0) {
+        // if we get here, the lower bound is the minimum value in the index.
+        // we can create a special bound only consisting of the index object id minus one.
+        objectIdStart--;
```

Comment:
What happens if we are unlucky enough to get two indices with sequential object IDs? Then doesn't this lower bound include all the entries from the index with the lesser ID?

Reply:
I think sequential IDs never happen, however I think `objectIdStart` without decreasing should work too?

Reply:
and actually I think this should be implemented differently, isn't this the exact case hit in https://github.com/arangodb/arangodb/blob/844b61ceff175a05f67339a9433317b35598b629/arangod/RocksDBEngine/RocksDBVPackIndex.cpp#L955 ?
We can probably avoid building the search slice entirely

Reply:
> I think sequential IDs never happen, however I think objectIdStart without decreasing should work too?
I think the lower bound is always inclusive  even in the reverse case

Reply:
@graetzer : That is probably true. I am happy if you reuse this branch for improving it, or if you create a new one with you suggestions. Up to you...